docs/models/model-typing.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: Typing your Model (TypeScript)
+sidebar_position: 8
+---
+
+# Typing your Model with TypeScript
+
+## Automated Attribute Typing
+
+Model methods are strictly typed based on your attributes, but require you to define you to write a bit of configuration first.
+`Model` accept two generic types that you can use to let it know what its Attributes & Creation Attributes are like:
+
+You can use [`InferAttributes`](pathname:///api/v7/index.html#InferAttributes), 
+and [`InferCreationAttributes`](pathname:///api/v7/index.html#InferCreationAttributes) to define them. They will extract Attribute typings
+directly from the Model:
+
+```typescript
+import { Attribute, PrimaryKey, AutoIncrement, NotNull } from '@sequelize/core/decorators-legacy';
+import { Model, InferAttributes, InferCreationAttributes, CreationOptional } from '@sequelize/core';
+
+// order of InferAttributes & InferCreationAttributes is important.
+export class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+  @Attribute(DataTypes.INTEGER)
+  @PrimaryKey
+  @AutoIncrement
+  // 'CreationOptional' is a special type that marks the attribute as optional
+  // when creating an instance of the model (such as using Model.create()).
+  declare id: CreationOptional<number>;
+  
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  declare name: string;
+}
+
+// TypeScript will now be able to catch errors such as this typo:
+// error-next-line
+await User.create({ nAme: 'john' });
+```
+
+Important things to know about [`InferAttributes`](pathname:///api/v7/index.html#InferAttributes) & [`InferCreationAttributes`](pathname:///api/v7/index.html#InferCreationAttributes) work: They will select all declared properties of the class except:
```

Comment:
```suggestion
Important things to know about how [`InferAttributes`](pathname:///api/v7/index.html#InferAttributes) & [`InferCreationAttributes`](pathname:///api/v7/index.html#InferCreationAttributes) work  - they will select all declared properties of the class, except the following:
```

Look at this again - word missing? I do not understand the double colon in this line.